libraries/ui-library/src/components/six-select/util.ts

```diff
@@ -0,0 +1,64 @@
+export interface MenuItem {
+  value: string;
+}
+
+export function getValue(value: unknown, multiple: boolean, menuItems: MenuItem[]): string | string[] {
+  if (multiple) {
+    if (Array.isArray(value)) {
+      return menuItems.filter((item) => value.includes(item.value)).map((item) => item.value);
+    } else if (typeof value === 'string') {
+      return menuItems.filter((item) => value === item.value).map((item) => item.value);
+    } else {
+      return [];
+    }
+  } else {
+    if (typeof value === 'string') {
+      return menuItems.find((item) => value === item.value)?.value ?? '';
+    } else {
+      return '';
+    }
+  }
+}
+
+export function isValidValue(value: unknown, multiple: boolean, menuItems: MenuItem[]): boolean {
+  if (multiple) {
+    if (Array.isArray(value)) {
+      if (value.length === 0) {
+        return true;
+      }
+      return menuItems.filter((item) => value.includes(item.value)).map((item) => item.value).length == value.length;
+    } else if (typeof value === 'string') {
+      if (value === '') {
+        return true;
+      }
+      return menuItems.filter((item) => value === item.value).map((item) => item.value).length > 0;
+    } else {
+      return false;
+    }
+  }
+
+  if (typeof value === 'string') {
+    if (value === '') {
+      return true;
+    }
+    return menuItems.find((item) => value === item.value) != null;
+  } else {
+    return false;
+  }
+}
```

Comment:
Same goes for this one, the nesting decreases code readability. However, I don't see a quick refactoring which would improve this. Maybe at least unwrap the unnecessary `else`s and extract `(item) => item.value` function (could even be extracted to outer scope, since it's also used in the other function.

Reply:
Tried to simplify it: ccf3feb0fbf21f40a5202a2489eff028507c35b5